example/src/main/java/io/netty/example/stomp/websocket/StompChatHandler.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.example.stomp.websocket;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.DecoderResult;
+import io.netty.handler.codec.stomp.DefaultStompFrame;
+import io.netty.handler.codec.stomp.StompCommand;
+import io.netty.handler.codec.stomp.StompFrame;
+
+import java.nio.CharBuffer;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.netty.buffer.ByteBufUtil.*;
+import static io.netty.handler.codec.stomp.StompHeaders.*;
+import static io.netty.util.CharsetUtil.*;
+
+@Sharable
+public class StompChatHandler extends SimpleChannelInboundHandler<StompFrame> {
+
+    static final StompChatHandler INSTANCE = new StompChatHandler();
+
+    private final ConcurrentHashMap<String, Set<StompSubscription>> CHAT_DESTINATIONS =
+            new ConcurrentHashMap<String, Set<StompSubscription>>();
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, StompFrame inboundFrame) throws Exception {
+        DecoderResult decoderResult = inboundFrame.decoderResult();
+        if (decoderResult.isFailure()) {
+            sendErrorFrame("rejected frame", decoderResult.toString(), ctx);
+            return;
+        }
+
+        switch (inboundFrame.command()) {
+        case STOMP:
+        case CONNECT:
+            onConnect(inboundFrame, ctx);
+            break;
+        case SUBSCRIBE:
+            onSubscribe(inboundFrame, ctx);
+            break;
+        case SEND:
+            onSend(inboundFrame, ctx);
+            break;
+        case UNSUBSCRIBE:
+            onUnsubscribe(inboundFrame, ctx);
+            break;
+        case DISCONNECT:
+            onDisconnect(inboundFrame, ctx);
+            break;
+        default:
+            sendErrorFrame("unsupported command",
+                           "Received unsupported command " + inboundFrame.command(), ctx);
+        }
+    }
+
+    private void onConnect(StompFrame inboundFrame, ChannelHandlerContext ctx) {
```

Comment:
static

Reply:
Done

Reply:
seems like you didn't push this yet

Reply:
Yes just pushed, also some fixes added in `StompWebSocketClientPageHandler.java`